server.go

```diff
@@ -22,6 +23,10 @@ const idleWorkerTimeout = 10 * time.Second
 // Maximum number of workers
 const maxWorkersCount = 10000
 
+// aLongTimeAgo is a non-zero time, far in the past, used for
+// immediate cancelation of network operations.
+var aLongTimeAgo = time.Unix(1, 0)
```

Comment:
If your system time is set to 1970 this won't work. I know this sounds a bit far-fetched, but maybe we should set it at least earlier than epoch? like 1965 or someting?

Reply:
This is the value used by the standard library: https://github.com/golang/go/blob/43f54c8d2e3bddfc6ad7887286eb6564986cb6ad/src/net/http/http.go#L21-L23.